### PR TITLE
[Ready for Review] Nerfs IPCs Part 1/1,034

### PR DESCRIPTION
**Pull request by LordFowl**
Refactors electrocute_act slightly. Electricity will now only start in your hands if ground_zero is explicitly set to l_hand or r_hand. All instances where electrocute_act is called because you touched something (IE opening a crate, touching the powergrid) have been set so that ground_zero is your currently active hand.

Otherwise, ground_zero will be randomly selected from available organs.

This is important because it is the siemens_coefficient of ground_zero **only** that affects electrical conductivity. EG if you get tesla_zapped in the chest you will not be saved by wearing insulated gloves. Once the electricity is in your body it does not matter.

Sufficiently powerful electricity (shock_damage >= 6) will induce an EMP in the relevant contact zones. This EMP will affect all items in the relevant organ only. Shock damage will still become reduced as the arc propagates through your body, and the EMP's produced will be updated accordingly.

The IPC power cell organ will now produce effects when EMP'd based on the current damage value of the organ pre-decrement, ranging from stuttering and blurriness to unconsciousness. Other special EMP effects for other IPC organs are pending, but I am thinking of holding it off for Part 2/1,034

Baton class weapons have been modified. Their raw force damage has been reduced, but they will now deal shock damage to a roughly equivalent value.

Harmbatons will deal 5 brute and 10 shock, and their electrocute_act will have a defined def zone (e.g it is a localized shock and there will be no arcs)

Cattleprods will deal 3 brute and 6 shock on both harm and help intents, and their electrocute_act is non-localized and will cause arcing.

Stunrods will deal 7 brute and 14 shock on both harm and help intents, and their electrocute_act is non-localized and will cause arcing.

(For clarification, cattleprods and stunrods currently still deal no brute on help intent, but will cause shock damage)

Fixes an issue with electrocute_act where if def_zone is called would not actually do anything.

- [x] Changelog
- [x] Fix EMP propagation in affected cells
- [x] Testing
- [x] Balance

### CODE REVIEW

**Comment by BurgerLUA** [comment]
> Sufficiently powerful electricity (shock_damage >= 6) will induce an EMP in the relevant contact zones. This EMP will affect all items in the relevant organ only. Shock damage will still become reduced as the arc propagates through your body, and the EMP's produced will be updated accordingly.

How will this affect rounds with shitty malf AIs that do nothing but shock doors?

**Comment by LordFowl** [comment]
You will get shocked, and every item in the organs that get shocked will have an appropriate emp_act called.

File: code/game/objects/items/weapons/stunbaton.dm [hunk 1]
```diff
@@ -121,6 +122,10 @@
 		stun *= 0.5
 		if(status)		//Checks to see if the stunbaton is on.
 			agony *= 0.5	//whacking someone causes a much poorer contact than prodding them.
+			if(sheathed) //however breaking the skin results in a more potent electric shock or some bullshit. im a coder, not a doctor
```
Comment (Lohikar) [inline code/game/objects/items/weapons/stunbaton.dm]: `whacking them produces a poorer contact, but a better contact`

Comment (LordFowl) [reply code/game/objects/items/weapons/stunbaton.dm]: im a coder not a doctor, and i can't take responsibility for the stupidity of those who came before me. some quick pseudoscience:

a direct, non-breaking contact has better distribution of electromagic that allows the target to be more routinely stunned.

contact that breaks the skin is less uniform and less localized, but distributes across the entire body's interior 'n shit. 

fuck, players can't even read these comments, i ain't gotta explain shit.

Comment (OneOneThreeEight) [reply code/game/objects/items/weapons/stunbaton.dm]: >fuck, players can't even read these comments, i ain't gotta explain shit.

we can and you do :)

**Comment by K0NFL1QT** [comment]
Making stun batons do actual electric shocks is probably a bad idea, because then they're by default a lethal weapon and using them for detainment is not only excessive force in character, but risks causing IPCs and organics lasting damage, or potentially fatal infections. And that's not even considering the additional arcing you've added to the improvised stun weapons.

**Comment by LordFowl** [comment]
Stun batons only deal electric damage in harm mode, where they were already a lethal weapon.

**Comment by OneOneThreeEight** [comment]
Why wouldn't it deal electric damage in disarm mode? The motions that are different between disarm and harm intent is the difference between a prod and a golf swing. I'm not sure if doing a swing motion with the stun end would cause it to be more electrically effective rather than physically effective, as it's still discharging the same energy.

**Comment by LordFowl** [comment]
It does not break the skin in disarm mode, indeed it deals no damage except agony damage, which is electric shock but in a lesser disagree.

**Comment by CakeIsOssim** [comment]
So, we're not going to see synth limbs get struck with a 10 unit shock and have it achieve the same effect as an ion rifle due to EMP?

**Comment by LordFowl** [comment]
A 10 unit shock should deal EMP to the contact zone organ but then be reduced to a 4 unit shock for the next arc, and then a 1 unit shock, and then a 1/4 unit shock and so on. Shock less than 6 produces no EMP, and even a 10 unit shock produces a very weak EMP.

**Comment by CakeIsOssim** [comment]
This is an approved™ IPC nerf.

**Pull request merged**